container-search/src/test/java/com/yahoo/search/handler/test/JSONSearchHandlerTestCase.java

```diff
@@ -0,0 +1,476 @@
+        assertOkResult(driver.sendRequest(uri, com.yahoo.jdisc.http.HttpRequest.Method.POST, json.toString(), JSON_CONTENT_TYPE), tiledResult);
+    }
+
+    private static final String pageResult =
+            "<?xml version=\"1.0\" encoding=\"utf-8\" ?>\n" +
+                    "<page version=\"1.0\">\n" +
+                    "\n" +
+                    "  <content>\n" +
+                    "    <hit relevance=\"1.0\">\n" +
+                    "      <id>testHit</id>\n" +
+                    "      <uri>testHit</uri>\n" +
+                    "    </hit>\n" +
+                    "  </content>\n" +
+                    "\n" +
+                    "</page>\n";
+
+    private void assertPageResult(JSONObject json, RequestHandlerTestDriver driver) throws Exception {
+        assertOkResult(driver.sendRequest(uri, com.yahoo.jdisc.http.HttpRequest.Method.POST, json.toString(), JSON_CONTENT_TYPE), pageResult);
+    }
+
+    private void assertOkResult(RequestHandlerTestDriver.MockResponseHandler response, String expected) {
+        assertEquals(expected, response.readAll());
+        assertEquals(200, response.getStatus());
+        assertEquals(selfHostname, response.getResponse().headers().get(myHostnameHeader).get(0));
+    }
+
+
+    private RequestHandlerTestDriver driverWithConfig(String configDirectory) throws Exception {
+        IOUtils.copyDirectory(new File(testDir, configDirectory), new File(tempDir), 1);
+        generateComponentsConfigForActive();
+        configurer.reloadConfig();
+
+        SearchHandler newSearchHandler = fetchSearchHandler(configurer);
+        assertTrue("Do I have a new instance of the search handler?", searchHandler != newSearchHandler);
+        return new RequestHandlerTestDriver(newSearchHandler);
+    }
+
+
+
+    private void createRequestMapping(Inspector inspector, Map<String, String> map, String parent){
+        inspector.traverse((ObjectTraverser) (key, value) -> {
+            String qualifiedKey = parent + key;
+            switch (value.type()) {
+                case BOOL:
+                    map.put(qualifiedKey, Boolean.toString(value.asBool()));
+                    break;
+                case DOUBLE:
+                    map.put(qualifiedKey, Double.toString(value.asDouble()));
+                    break;
+                case LONG:
+                    map.put(qualifiedKey, Long.toString(value.asLong()));
+                    break;
+                case STRING:
+                    map.put(qualifiedKey , value.asString());
+                    break;
+                case OBJECT:
+                    if (key.equals("grouping")) {
+                        createRequestMapping(value, map, "");
+                    } else {
+                        createRequestMapping(value, map, qualifiedKey+".");
+                        break;
+                    }
+            }
+
+        });
+    }
```

Comment:
You must increase the visibility of the production code and test that, rather than copy the code and test the copy ;) Package private is OK.